File: inflearn-core-spring-security-project/src/main/java/com/example/inflearncorespringsecurityproject/repository/UserRepository.java
package com.example.inflearncorespringsecurityproject.repository;

import com.example.inflearncorespringsecurityproject.domain.entity.Account;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserRepository extends JpaRepository<Account, Long> {

    Account findByUsername(String username);

    int countByUsername(String username);
}

File: library-app/src/main/kotlin/com/example/libraryapp/domain/user/UserRepository.kt
package com.example.libraryapp.domain.user

import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository : JpaRepository<User, Long>, UserRepositoryCustom {

    fun findByName(name: String): User?

//    @Query("select distinct u from User u left join fetch u.userLoanHistories")
//    fun findAllWithHistories(): List<User>
}